server/ServerNetworking.cpp

```diff
@@ -165,6 +165,25 @@ PlayerConnection::PlayerConnection(boost::asio::io_context& io_context,
     m_disconnected_callback(disconnected_callback)
 {}
 
+namespace {
+    struct AsyncCloseClosure {
+        AsyncCloseClosure(boost::optional<boost::asio::ip::tcp::socket> &&socket, int id) :
+            m_socket(std::move(socket)),
+            m_id(id)
+        {}
+
+        void operator()() {
+            TraceLogger(network) << "Asynchronously closing socket for player id " << m_id;
+            if (m_socket)
+                m_socket->close();
+            TraceLogger(network) << "Socket closed for player id " << m_id;
```

Comment:
I just meant that if `m_socket` was empty, then the log saying that the socket was closed is misleading... since nothing was closed.

Reply:
Made `m_socket` test around log calls.